x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/common/validation/SourceDestValidator.java

```diff
@@ -0,0 +1,513 @@
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+import static org.elasticsearch.cluster.metadata.MetaDataCreateIndexService.validateIndexOrAliasName;
+
+/**
+ * Validation of source indexes and destination index in different situations (preview, put)
+ */
+public final class SourceDestValidator {
+
+    // messages
+    public static final String SOURCE_INDEX_MISSING = "Source index [{0}] does not exist";
+    public static final String SOURCE_LOWERCASE = "Source index [{0}] must be lowercase";
+    public static final String DEST_IN_SOURCE = "Destination index [{0}] is included in source expression [{1}]";
+    public static final String DEST_LOWERCASE = "Destination index [{0}] must be lowercase";
+    public static final String NEEDS_REMOTE_CLUSTER_SEARCH = "Source index is configured with a remote index pattern(s) [{0}]"
+        + " but the current node [{1}] is not allowed to connect to remote clusters."
+        + " Please enable cluster.remote.connect for all data nodes.";
+    public static final String ERROR_REMOTE_CLUSTER_SEARCH = "Error during resolving remote source, error: {0}";
+    public static final String UNKNOWN_REMOTE_CLUSTER_LICENSE = "Error during license check ({0}) for remote cluster "
+        + "alias(es) {1}, error: {2}";
+    public static final String FEATURE_NOT_LICENSED_REMOTE_CLUSTER_LICENSE = "License check failed for remote cluster "
+        + "alias [{0}], at least a [{1}] license is required, found license [{2}]";
+    public static final String REMOTE_CLUSTER_LICENSE_INACTIVE = "License check failed for remote cluster "
+        + "alias [{0}], license is not active";
+    public static final String TIMEOUT_CHECK_REMOTE_CLUSTER_LICENSE = "Timeout during license check ({0}) for remote cluster "
+        + "alias(es) [{1}]";
+
+    static class Context {
+        private final ClusterState state;
+        private final IndexNameExpressionResolver indexNameExpressionResolver;
+        private final RemoteClusterService remoteClusterService;
+        private final RemoteClusterLicenseChecker remoteClusterLicenseChecker;
+        private final String[] source;
+        private final String dest;
+        private final String nodeName;
+        private final String license;
+
+        private ValidationException validationException = null;
+        private Set<String> resolvedSource = null;
+        private Set<String> resolvedRemoteSource = null;
+        private String resolvedDest = null;
+
+        Context(
+            final ClusterState state,
+            final IndexNameExpressionResolver indexNameExpressionResolver,
+            final RemoteClusterService remoteClusterService,
+            final RemoteClusterLicenseChecker remoteClusterLicenseChecker,
+            final String[] source,
+            final String dest,
+            final String nodeName,
+            final String license
+        ) {
```

Comment:
There is a number of lines in this PR where the closing parenthesis is in a new line.

Reply:
This is the new spotless enforce format...I personally don't like it at all :/